flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -138,6 +138,14 @@
 	 */
 	private boolean enableCommitOnCheckpoints = true;
 
+	/**
+	 * User-set flag to disable filter restored partitions with current
+	 * discovered partitions. It's enabled by default since it'll result unexpected
+	 * behaviors - e.g. When changing the topic name, or remove some topics,
+	 * The removed/renamed partitions will be still consumed.
+	 */
+	private boolean filterRestoredPartitionsWithDiscovered = true;
```

Comment:
If we make `true` as default value, is the behavior different from previous?

Reply:
Yep. As discussed on the maillist with @tzulitai , we decided to go with filter by default. Since this can cause unexpected behaviors in production when changing topic/remove partitions. But still leaving the possibility of maintaining the default behavior.

https://lists.apache.org/thread.html/245779f914f0f1d4a3fc7dfae04f2fbbc7edd682df4b6e0592a6d9d1@%3Cdev.flink.apache.org%3E